frontend/src/metabase/admin/settings/app/components/SettingsEditor/SettingsEditor.jsx

```diff
@@ -92,27 +99,29 @@ class SettingsEditor extends Component {
 
     // TODO: mutation bad!
     setting.value = newValue;
+
+    const handlerParams = [
+      oldValue,
+      newValue,
+      settingValues,
+      this.handleChangeSetting,
+    ];
+
     try {
       if (setting.onBeforeChanged) {
-        await setting.onBeforeChanged(
-          oldValue,
-          newValue,
-          settingValues,
-          this.handleChangeSetting,
-        );
+        await setting.onBeforeChanged(...handlerParams);
       }
 
       if (!setting.disableDefaultUpdate) {
         await updateSetting(setting);
       }
 
       if (setting.onChanged) {
-        await setting.onChanged(
-          oldValue,
-          newValue,
-          settingValues,
-          this.handleChangeSetting,
-        );
+        await setting.onChanged(...handlerParams);
+      }
+
+      if (options?.onChanged) {
+        await options.onChanged(...handlerParams);
```

Comment:
I like it. I also like that it doesn't overwrite the `onChanged` in the setting itself 👍